examples/fuse_examples/imaging/classification/knight/make_predictions_file.py

```diff
@@ -80,67 +88,94 @@ def make_predictions_file(
+    pl_module = LightningModuleDefault(
+        model_dir=model_dir,
+        model=model,
+    )
+
+    pl_module.set_predictions_keys([predictions_key_name])
+
+    pl_trainer = pl.Trainer(
+        default_root_dir=model_dir,
+        accelerator="gpu",
+        devices=1,
+        strategy=None,
+        auto_select_gpus=True,
     )
 
+    predictions = pl_trainer.predict(pl_module, dl, ckpt_path=checkpoint)
+
     # Convert to required format
     if task_num == 1:
         class_names = TASK1_CLASS_NAMES
     elif task_num == 2:
         class_names = TASK2_CLASS_NAMES
     else:
         raise Exception(f"Unexpected task num {task_num}")
+
     predictions_score_names = [f"{cls_name}-score" for cls_name in class_names]
-    predictions_df[predictions_score_names] = pd.DataFrame(
-        predictions_df[predictions_key_name].tolist(), index=predictions_df.index
-    )
+    data = []
+    for prediction in predictions:
+        for i in range(len(prediction["id"])):
+            data.append(list(prediction["model.output.head_0"][i]))
+
+    predictions_df = pd.DataFrame(data, columns=list(predictions_score_names))
     predictions_df.reset_index(inplace=True)
-    predictions_df.rename({"index": "case_id"}, axis=1, inplace=True)
-    predictions_df = predictions_df[["case_id"] + predictions_score_names]
+    predictions_df.rename({"index": "Case_id"}, axis=1, inplace=True)
 
     # save file
     save_dataframe(predictions_df, output_filename, index=False)
 
 
 if __name__ == "__main__":
     """
-    Automaitically make prediction files in the requested format - given path to model dir create by FuseMedML during training
-    Usage: python make_predictions_file <model_dir> <checkpint> <data_path> <cache_path> <split_path> <output_filename> <predictions_key_name> <task_num>.
-    See details in function הmake_predictions_file.
+    Automaitically make prediction files in the requested format - given model definition and path to model dir create by FuseMedML during training
     """
-    if len(sys.argv) == 1:
-        # no arguments - set arguments inline - see details in function make_predictions_file
-        model_dir = ""
-        checkpoint = "best"
-        data_path = ""
-        cache_path = ""
-        split = f"{pathlib.Path(__file__).parent.resolve()}/baseline/splits_final.pkl"
-        output_filename = "validation_predictions.csv"
-        predictions_key_name = "model.output.head_0"
-        task_num = 1  # 1 or 2
+    # no arguments - set arguments inline - see details in function make_predictions_file
+    model_dir = ""
+    checkpoint = "best"
+    data_path = ""
+    cache_path = ""
+    split = None
+    output_filename = "validation_predictions.csv"
+    predictions_key_name = "model.output.head_0"
+    task_num = 1  # 1 or 2
+
+    use_data = {"imaging": True, "clinical": True}  # specify whether to use imaging, clinical data or both
+    num_classes = 2
+
+    target_name = "data.gt.gt_global.task_1_label"
```

Comment:
Instead of copy-pasting, can you please wrap it with a function called here in your training process?

Reply:
actually I don't see any update needed in the KNIGHT README file.